pkg/occlient/occlient_test.go

```diff
@@ -3774,3 +3774,121 @@ func Test_updateEnvVar(t *testing.T) {
 		})
 	}
 }
+
+func Test_findContainer(t *testing.T) {
+	type args struct {
+		name       string
+		containers []corev1.Container
+	}
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Case 1 - Find the container",
+			args: args{
+				name: "foo",
+				containers: []corev1.Container{
+					{
+						Name: "foo",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								MountPath: "/tmp",
+								Name:      "test-pvc",
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name: "Case 2 - Error if container not found",
+			args: args{
+				name: "foo2",
+				containers: []corev1.Container{
+					{
+						Name: "foo",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								MountPath: "/tmp",
+								Name:      "test-pvc",
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Case 3 - Error when passing in blank container name",
+			args: args{
+				name: "",
+				containers: []corev1.Container{
+					{
+						Name: "foo",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								MountPath: "/tmp",
+								Name:      "test-pvc",
+							},
+						},
+					},
+				},
+			},
+			wantErr: true,
+		},
+		{
+			name: "Case 4 - Check against multiple containers (rather than one)",
+			args: args{
+				name: "foo",
+				containers: []corev1.Container{
+					{
+						Name: "bar",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								MountPath: "/tmp",
+								Name:      "test-pvc",
+							},
+						},
+					},
+					{
+						Name: "foo",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								MountPath: "/tmp",
+								Name:      "test-pvc",
+							},
+						},
+					},
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			// Run function findContainer
+			container, err := findContainer(tt.args.containers, tt.args.name)
+
+			// Check that the container matches the name
+			if err == nil && container.Name != tt.args.name {
+				t.Errorf("Wrong container returned, wanted container %v, got %v", tt.args.name, container.Name)
+			}
+
+			// Error checking findContainer
+			if !tt.wantErr == (err != nil) {
```

Comment:
This condition is already satisfied by the below two conditions, so the error is thrown twice (once by this condition and the other from below
```
--- FAIL: Test_findContainer/Case_1_-_Find_the_container (0.00s)
        occlient_test.go:3883:  client.findContainer() unexpected error <nil>, wantErr true
        occlient_test.go:3887: test failed, expected: false, got true
```